spec/rubocop/cop/rspec/no_let_spec.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::RSpec::NoLet do
+  subject(:cop) { described_class.new(config) }
+
+  let(:config) { RuboCop::Config.new }
+
+  # TODO: Write test code
+  #
+  # For example
+  it 'registers an offense when using `#let`' do
+    expect_offense(<<~RUBY)
+      let(:foo) { Foo.new }
+      ^^^^^^^^^ Avoid using `let` and `subject` – use a method call or local variable instead.
+    RUBY
+  end
+
+  it 'registers an offense when using `#subject` without name' do
+    expect_offense(<<~RUBY)
+      subject { Foo.new }
+      ^^^^^^^ Avoid using `let` and `subject` – use a method call or local variable instead.
+    RUBY
+  end
+
+  it 'registers an offense when using `#subject` with name' do
+    expect_offense(<<~RUBY)
+      subject(:foo) { Foo.new }
+      ^^^^^^^^^^^^^ Avoid using `let` and `subject` – use a method call or local variable instead.
+    RUBY
+  end
+
+  it 'does not register an offense when using `#before`' do
+    expect_no_offenses(<<~RUBY)
+      before { foo }
+    RUBY
+  end
+
+  context 'when using AllowSubject configuration', :config do
+    subject(:cop) { described_class.new(config) }
+
+    let(:cop_config) do
+      { 'AllowSubject' => true }
+    end
+
+    it 'registers an offense when using `#let`' do
+      expect_offense(<<~RUBY)
+        let(:foo) { Foo.new }
+        ^^^^^^^^^ Avoid using `let` and `subject` – use a method call or local variable instead.
+      RUBY
+    end
+
+    it 'does not register an offense when using `#subject` without a name' do
```

Comment:
Minor: most often 'flags'/'ignores' are used as prefixes, but this is actually quite diverse.